pkg/strategy/grid2/strategy.go

```diff
@@ -1414,34 +1416,81 @@ func (s *Strategy) recoverGridWithOpenOrdersByScanningTrades(ctx context.Context
 	// 3. get the filled orders from pin-order map
 	filledOrders := pinOrdersFilled.AscendingOrders()
 	numsFilledOrders := len(filledOrders)
-	i := 0
 	if numsFilledOrders == int(expectedOrderNums-openOrdersOnGridNums) {
 		// nums of filled order is the same as Size - 1 - num(open orders)
 	} else if numsFilledOrders == int(expectedOrderNums-openOrdersOnGridNums+1) {
-		i++
+		filledOrders = filledOrders[1:]
 	} else {
 		return fmt.Errorf("not reasonable num of filled orders")
 	}
 
-	// 4. emit the filled orders
+	// 4. verify the grid
+	if err := s.verifyFilledGrid(s.grid.Pins, pinOrdersOpen, filledOrders); err != nil {
```

Comment:
I think you can move the verifyFilledGrid method to the Grid object.

so that you can call grid.Verify() ?

Reply:
And which should make the test easier.

Reply:
I will move to grid's method after change pinOrderMap from string key to fixedpoint.Value key